File: python/heliosops/tests/services/contracts_test.py
import unittest

from shared.contracts.contracts import CONTRACTS


class ContractTests(unittest.TestCase):
    def test_contracts_expose_required_keys(self) -> None:
        self.assertEqual(CONTRACTS["gateway"]["id"], "gateway")
        self.assertIsInstance(CONTRACTS["routing"]["port"], int)


if __name__ == "__main__":
    unittest.main()

File: ruby/mercuryledger/tests/services/contracts_test.rb
# frozen_string_literal: true

require_relative '../test_helper'
require_relative '../../shared/contracts/contracts'

class ContractsTest < Minitest::Test
  def test_contracts_expose_required_keys
    contracts = MercuryLedger::Contracts::CONTRACTS
    assert_equal 'gateway', contracts[:gateway][:id]
    assert_equal 8111, contracts[:routing][:port]
  end
end
